Sources/Parsing/Parsers/Float.swift

```diff
@@ -0,0 +1,172 @@
+extension BinaryFloatingPoint where Self: LosslessStringConvertible {
```

Comment:
Worth noting that we need to abstract over `BinaryFloatingPoint` _and_ `LosslessStringConvertible` here. While `FixedWidthInteger` inherits from it automatically, `BinaryFloatingPoint` does not.